src/pykeen/models/baseline.py

```diff
@@ -0,0 +1,464 @@
+    scores = sklearn_normalize(scores, norm="l1")
+
+    # note: we need to work with dense arrays only to comply with returning torch tensors. Otherwise, we could
+    # stay sparse here, with a potential of a huge memory benefit on large datasets!
+    return torch.from_numpy(scores.todense())
+
+
+class MarginalDistributionBaseline(EvaluationOnlyModel):
+    r"""
+    Score based on marginal distributions.
+
+    To predict scores for the tails, we simplify
+
+    .. math ::
+        P(t | h, r) = P(t | h) * P(t | r)
+
+    Depending on the settings, we either set P(t | *) = 1/n, or estimate them by counting occurrences in the training
+    triples.
+
+    .. note ::
+        This model cannot make use of GPU acceleration, since internally it uses scipy's sparse matrices.
+    """
+
+    def __init__(
+        self,
+        triples_factory: CoreTriplesFactory,
+        entity_margin: bool = True,
+        relation_margin: bool = True,
+    ):
+        """
+        Initialize the model.
+
+        :param triples_factory:
+            The triples factory containing the training triples.
+        """
+        super().__init__(
+            triples_factory=triples_factory,
+            random_seed=0,  # TODO: Why do we provide the random seed?
+            preferred_device='cpu',
+        )
+        h, r, t = numpy.asarray(triples_factory.mapped_triples).T
+        if relation_margin:
+            self.head_per_relation, self.tail_per_relation = [
+                get_csr_matrix(
+                    row_indices=r,
+                    col_indices=col_indices,
+                    shape=(triples_factory.num_relations, triples_factory.num_entities),
+                )
+                for col_indices in (h, t)
+            ]
+        else:
+            self.head_per_relation = self.tail_per_relation = None
+        if entity_margin:
+            self.head_per_tail, self.tail_per_head = [
+                get_csr_matrix(
+                    row_indices=row_indices,
+                    col_indices=col_indices,
+                    shape=(triples_factory.num_entities, triples_factory.num_entities),
+                )
+                for row_indices, col_indices in ((t, h), (h, t))
+            ]
+        else:
+            self.head_per_tail = self.tail_per_head = None
+
+    def score_t(self, hr_batch: torch.LongTensor) -> torch.FloatTensor:  # noqa:D102
+        return _score(
+            entity_relation_batch=hr_batch,
+            per_entity=self.tail_per_head,
+            per_relation=self.tail_per_relation,
+            num_entities=self.num_entities,
+        )
+
+    def score_h(self, rt_batch: torch.LongTensor) -> torch.FloatTensor:  # noqa:D102
+        return _score(
+            entity_relation_batch=rt_batch.flip(1),
+            per_entity=self.head_per_tail,
+            per_relation=self.head_per_relation,
+            num_entities=self.num_entities,
+        )
+
+
+# TODO: Remove this model from this PR?
```

Comment:
@cthoyt any comments on this? I think it might be worth to split this PR into two parts, where one only adds the `MarginalDistributionBaseline`, and keeps out the inverse triple one for now.

For the inverse triple one there is still some work to be done for reducing code duplication with the leakage detection.

Reply:
I'd be okay with splitting the PR so we have something done and merged first